stellargraph/layer/gcn_lstm.py

```diff
@@ -244,6 +247,19 @@ def __init__(
         bias_regularizer=None,
         bias_constraint=None,
     ):
+        if generator is not None:
+            if not isinstance(generator, SlidingFeaturesNodeGenerator):
+                raise ValueError(
+                    f"generator: expected a SlidingFeaturesNodeGenerator, found {type(generator).__name__}"
+                )
+
+            if seq_len is not None or adj is not None:
+                raise ValueError(
+                    "expected only one of generator and (seq_len, adj) to be specified, found multiple"
+                )
+
+            adj = generator.graph.to_adjacency_matrix(weighted=True).todense()
+            seq_len = generator.window_size
```

Comment:
Given that this class is still in experimental, maybe this is a good opportunity to rename `seq_len` to be `window_size` for consistency?

Reply:
I'd prefer not to "contaminate" this PR too many with things that may require changes elsewhere (e.g. in the demo, and/or merge conflicts with #1560). In addition, I think there's a bit more work required here to make this class super clean, e.g.:

- switching the `generator` parameter to be more usable (making `seq_len` etc. optional)
- potentially even removing the `seq_len` parameters and just using `generator`